private/rules/pom_file.bzl

```diff
@@ -1,33 +1,41 @@
 load(":has_maven_deps.bzl", "MavenInfo", "has_maven_deps")
 
-_PLAIN_DEP = """        <dependency>
-            <groupId>{0}</groupId>
-            <artifactId>{1}</artifactId>
-            <version>{2}</version>
-        </dependency>"""
+def _format_dep(unpacked):
+    return "".join([
+ "        <dependency>\n",
+ "            <groupId>%s</groupId>\n" % unpacked.groupId,
+ "            <artifactId>%s</artifactId>\n" % unpacked.artifactId,
+ "            <version>%s</version>\n" % unpacked.version,
+("            <type>%s</type>\n" % unpacked.type) if unpacked.type and unpacked.type != "jar" else "",
```

Comment:
Does this work? I would expect: `"<type>%s</type>\n" % unpacked.type if unpacked.type and unpacked.type != "jar" else ""` That is, the closing bracket seems like it's in the wrong place (and it's confusing even if it's the right place :) )

Reply:
Also, the struct has a `type_` field, and you're using `type` here, so the code won't work. I'm not sure how this managed to sneak past the CI, but that's my problem to solve :)